mdata/cache/accnt/flat_accnt.go

```diff
@@ -0,0 +1,310 @@
+package accnt
+
+import (
+	"github.com/raintank/worldping-api/pkg/log"
+	"sort"
+	"time"
+)
+
+const evictQSize = 1000
+
+// it's easily possible for many events to happen in one request,
+// we never want this to fill up because otherwise events get dropped
+const eventQSize = 100000
+
+// Flat accounting
+//
+// Keeps track of the chunk cache size and in which order the contained
+// chunks have been used to last time. If it detects that the total cache
+// size is above the given limit, it feeds the least recently used
+// cache chunks into the evict queue, which will get consumed by the
+// evict loop.
+
+type FlatAccnt struct {
+	// total size of cache we're accounting for
+	total uint64
+
+	// metric accounting per metric key
+	metrics map[string]*FlatAccntMet
+
+	// the size limit, once this is reached we'll start evicting data
+	maxSize uint64
+
+	// a last-recently-used implementation that keeps track of all chunks
+	// and which hasn't been used for the longest time. the eviction
+	// function relies on this to know what to evict.
+	lru *LRU
+
+	// whenever a chunk gets evicted a job gets added to this queue. it is
+	// consumed by the chunk cache, which will evict whatever the jobs in
+	// evictQ tell it to
+	evictQ chan *EvictTarget
+
+	// a queue of add and hit events to be processed by the accounting.
+	// each add means data got added to the cache, each hit means data
+	// has been accessed and hence the LRU needs to be updated.
+	eventQ chan *FlatAccntEvent
```

Comment:
FlatAccntEvent contains a uint8 and a pointer. there's no need to use pointers to a FlatAccntEvent.
in fact, regarding the AddPayload and HitPayload, i suspect it may be better to put those values in the payload attribute as well, instead of pointers to them. since they are so small i suspect you'll be better off avoiding the dereferencing.